rtl/eclrtl/rtldynfield.cpp

```diff
@@ -1363,47 +1449,50 @@ class GeneralRecordTranslator : public CInterfaceOf<IDynamicTransform>
         }
     } *matchInfo;
 
-    size32_t estimateNewSize(const RtlRow &sourceRow) const
+    static size32_t translateScalarFromUtf8(ARowBuilder &builder, size32_t offset, const RtlFieldInfo *field, const RtlTypeInfo &destType, const RtlTypeInfo &sourceType, const char *source, size_t srcSize)
     {
-        //DBGLOG("Source record size is %d", (int) sourceRow.getRecordSize());
-        size32_t expectedSize = sourceRow.getRecordSize() - fixedDelta;
-        //DBGLOG("Source record size without omitted fixed size fields is %d", expectedSize);
-        ForEachItemIn(i, unmatched)
+        switch(destType.getType())
         {
-            unsigned fieldNo = unmatched.item(i);
-            expectedSize -= sourceRow.getSize(fieldNo);
-            //DBGLOG("Reducing estimated size by %d to %d for omitted field %d (%s)", (int) sourceRow.getSize(fieldNo), expectedSize, fieldNo, sourceRecInfo.queryName(fieldNo));
+        case type_boolean:
+        case type_int:
+        case type_swapint:
+        case type_packedint:
+        case type_filepos:
+        case type_keyedint:
+        {
+            __int64 res = rtlUtf8ToInt(srcSize, source);
```

Comment:
Should probably call rtlStrToInt8 - otherwise the length should be in codepoints rather than bytes.
(Not there seems to be a bug in rtlUtf8ToInt - it should be converting the length to bytes)

Reply:
will change.